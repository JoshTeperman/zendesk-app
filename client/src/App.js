import React, { Component } from 'react';
import './App.css';
import TicketList from './'
const apiModule = require('./APIhelper.js')


class App extends Component {
  constructor(){
    super()
    this.state = {
      ticketData: null,
    }
  }

  async componentDidMount() {
    const ticketData = await apiModule.getTickets()
    // console.log(ticketData)
    this.setState({ ticketData: ticketData })
  }

  display(){
    // const currentData = this.state.ticketData
    // console.log(currentData)
    // console.log('hello from inside display function')
    // this.setState({data: 'test'})
  }

  render(){
    return (
      <div className="app">
        <p>
          {this.state.ticketData}
        </p>
        <TicketList className="ticket-list"/>
      </div>
    );
  }
}

export default App;
