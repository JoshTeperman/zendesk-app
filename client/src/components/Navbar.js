import React from 'react'
const links = require('../images/zendesk-links.png')

const Navbar = () => {
  return(
    <div className="navbar">
      <h4 className="add">+ Add</h4>
      <img src={links} alt="navigation links"></img>
    </div>
  )
}

export default Navbar;