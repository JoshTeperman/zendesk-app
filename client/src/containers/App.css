.page-container {
  height: 100vh;
  background: lightblue;
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-template-rows: 50px 40px auto;
}

.navbar { 
  background: red;
  grid-area: 1 / 1 / 2 / 3; 
}

.sidebar { 
  background: blue;
  grid-area: 2 / 1 / 13 / 2; 
}

.header { 
  background: green;
  grid-area: 2 / 2 / 3 / 3; 
}

.tickets-table { 
  background: yellow;
  grid-area: 3 / 2 / 4 / 3; 
  display: block;
  overflow: auto; 
}