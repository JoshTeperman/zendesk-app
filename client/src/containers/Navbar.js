import React, { Component } from 'react'
import Links from '../components/Links'

class Navbar extends Component {
  render(){
    return(
      <div className="navbar">
        <Links />
      </div>
    )
  }
}

export default Navbar;