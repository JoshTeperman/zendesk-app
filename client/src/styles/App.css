@import './Header.css';
@import './SideNav.css';
@import './Navbar.css';
@import './Pagination.css';
@import './Sidebar.css';
@import './Table.css'; 
@import './DescriptionModal.css';
@import './Skeleton.css';
@import './ErrorScreen.css';



* {
  margin: 0px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif
}



.page-container {
  height: 100vh;
  /* background: lightblue; */
  display: grid;
  grid-template-columns: 55px 300px auto;
  grid-template-rows: 45px 70px auto 40px;
}


:root {
  /*  colours  */
  --charcoal: #2F3941;
  --darkgreen: #10363D;
  --palegreen: #4C7378;
  --lightgreen: #77A313;
  --gray: #E9EBED;
  --lightgray: #F8F9F9;;
  --medium-gray: #68737D;
  --darkgray: #2F3941;
  --orange: #E34F32;
}








