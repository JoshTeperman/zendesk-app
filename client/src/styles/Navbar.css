.navbar { 
  border-bottom: 1px solid var(--gray);
  grid-area: 1 / 2 / 2 / 4; 
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

.add {
  font-size: 12px;
  font-weight: 500;
  color: var(--charcoal);
}

.zendesk-links {
  height: 100%;
}
