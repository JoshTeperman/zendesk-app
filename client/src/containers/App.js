import React, { Component } from 'react';
import '../styles/App.css';
import SideNav from '../components/SideNav'
import Navbar from '../components/Navbar'
import Sidebar from '../components/Sidebar'
import Header from '../components/Header'
import Table from '../components/Table'
import Loading from '../components/Loading'
import ErrorScreen from '../components/ErrorScreen'
import Pagination from '../components/Pagination'
const helper = require('../APIHelper.js')


class App extends Component {
  constructor() {
    super()
    this.state = {
      currentTickets: undefined,
      totalTickets: null,
      pages: {},
      status: 'loading'
    }
  }

  updatePage = (response) => {
    if (response.error) {
      this.setState({
        status: response.error
      })
    } else {
      this.setState({ 
          currentTickets: response.ticketData,
          totalTickets: response.totalTickets,
          pages: response.pages,
          status: 'tickets'
      })
    }
  }

  async componentDidMount() {
    const response = await helper.getTickets()
    this.updatePage(response)
  }

  loadPage = async (url) => {
    const response = await helper.getPage(url)
    this.updatePage(response)
  }

  render(){
    const {currentTickets, totalTickets, pages, status} = this.state
    return(
        <div className="page-container">
          <Navbar />
          <SideNav />
          <Sidebar />
          {(status >= 400 || status === 'Server Down') && <ErrorScreen status={status}/>}
          {status === 'loading' && <Loading />}
          {currentTickets && <Header totalTickets={totalTickets}/>}
          {currentTickets && <Table currentTickets={currentTickets} />}
          <Pagination pages={pages} loadPage={this.loadPage}/>
        </div>
    )
  }
}

export default App;
