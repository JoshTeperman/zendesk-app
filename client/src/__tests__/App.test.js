import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from "enzyme";
import App from '../containers/App';
import Navbar from '../containers/Navbar';
// import Sidebar from '../components/Sidebar';
// import Header from '../components/Header';
// import mockTickets from '../mocks/tickets.js';

// --> Rendering page on load 

// test.todo('App renders correctly'), () => {
//     const wrapper = shallow(<App />)
//     expect(toJSON(wrapper)).toMatchSnapshot()
// }


// __Happy PATH:__

// 1) Page loads at root endpoint ('/tickets') 

// 2) App() renders the home page before calling any API data, which executes componentDoesMount()
// - assert that componentDoesMount() is called once
// - assert that the page loads and components display correctly
// - assert that the loading screen displays
// - (assert that any error in loading shows an error page)

// 3) Async function getTickets() is called
// - assert that getTickets() is called once


// 7) App.js getTickets() function receives array of new ticket objects and updates this.state
// - assert response object is an array of ticket objects
// - assert updatePage() is called once with argument response-- assert updatePage sets the State correctly (currentTickets, totalTickets, pages, pageHistory), possibly use mock data

// 8) App.js render() is called with updated state, rendering tickets the the page.
// - assert render() is called once
// - assert loading screen unmounts
// - assert tickets are rendered correctly
// - assert that header message displays the correct number of tickets

// 9) Pagination buttons display dynamically
// - assert pagination buttons disappear when pages.nextPage || pages.previous_page === null
// - assert pagination buttons appear when pages.nextPage || pages.previous_page === 'url'

// 10) Ticket display updates when page buttons are clicked
// - assert PageButton onClick method loadPage() is called once when clicked with argument link (url)
// - assert getTickets() is called once
// - assert updatePage() is called once, the state is changed, and render() updates the display

// 11) Loading a new page saves the page to pageHistory
// 12) Use pageHistory to load pages that have already been viewed

// 15) Hover on ticket shows ticket details