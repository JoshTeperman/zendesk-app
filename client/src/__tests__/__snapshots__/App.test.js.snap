// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App and all it's child components render correctly 1`] = `
<div
  className="page-container"
>
  <Navbar />
  <SideNav />
  <Sidebar />
  <Loading />
  <Pagination
    loadPage={[Function]}
    pages={Object {}}
  />
</div>
`;

exports[`App Component App renders correctly 1`] = `
<div
  className="page-container"
>
  <Navbar />
  <SideNav />
  <Sidebar />
  <Loading />
  <Pagination
    loadPage={[Function]}
    pages={Object {}}
  />
</div>
`;

exports[`App Component App state renders correctly 1`] = `undefined`;
